apps/finance/app/src/components/NewTransfer.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Field, TextInput, Text, Button, DropDown, theme } from '@aragon/ui'
+
+class NewTransfer extends React.Component {
+  state = {
+    selectedToken: 0,
+    recipient: '',
+    amount: -1,
+  }
+  handleSelectToken = index => {
+    this.setState({ selectedToken: index })
+  }
+  handleRecipientUpdate = event => {
+    this.setState({ recipient: event.target.value })
+  }
+  handleAmountUpdate = event => {
+    const value = event.target.value
+    const amount = value === '' ? 0 : parseInt(value, 10)
+    this.setState({
+      amount: !Number.isInteger(amount) || amount < 0 ? -1 : amount,
+    })
+  }
+  render() {
+    const { title, tokens } = this.props
+    const { recipient, amount, selectedToken } = this.state
+    return (
+      <div>
+        <h1>{title}</h1>
+        <Field label="Recipient">
+          <TextInput
+            onChange={this.handleRecipientUpdate}
+            value={recipient}
+            wide
+          />
+        </Field>
+        <AmountField>
+          <label>
+            <Text.Block color={theme.textSecondary} smallcaps>
+              Amount
+            </Text.Block>
+          </label>
+          <CombinedInput>
+            <TextInput
```

Comment:
What do you think about using `TextInput.Number` to make this `type="number"` instead? Maybe it would simplify things?